plugins/header_rewrite/resources.cc

```diff
@@ -87,6 +87,11 @@ Resources::gather(const ResourceIDs ids, TSHttpHookID hook)
         TSDebug(PLUGIN_NAME, "\tAdding TXN client response status resource");
         resp_status = TSHttpHdrStatusGet(bufp, hdr_loc);
       }
+      if (client_bufp && client_hdr_loc) {
```

Comment:
I'm not familiar with this area but this block doesn't seem to be correct. If I read the code correctly, `bufp` and `hdr_loc` point to headers that will be modified on current event (hook). Since this block is under `case TS_HTTP_SEND_RESPONSE_HDR_HOOK:`, those should point to response headers.

`TSHttpTxnClientRespGet` sets the right pointers but this block overwrites those with the ones for client request headers.

Removing this block fixes header_rewrite_l_value and header_rewrite_cond_ssn_txn_count at minimum.